### DISCUSSION TITLE
We will also need AWS Firewall for traffic regulation. Can we have just one pair Firewall instances hosted in a shared VPC and use it for all the 3 environments to reduce cost?

### DISCUSSION THREAD

**Discussion by marijakstrazdas** [gruntwork-io/knowledge-base#150] [Help] [answered]
We will also need AWS Firewall for traffic regulation. Can we have just one pair Firewall instances hosted in a shared VPC and use it for all the 3 environments to reduce cost?

**Comment by yorinasub17** [comment, marked as answer]
_NOTE: Please be advised that Gruntwork currently does not support AWS Firewall. We neither support it in the Reference Architecture, nor do we have a module to configure it._

AWS Network Firewalls currently does not support monitoring traffic across VPC peering. This is explicitly mentioned as an unsupported architecture in the [official AWS documentation](https://docs.aws.amazon.com/network-firewall/latest/developerguide/architectures.html). As such, it is currently not possible to share a network firewall across multiple VPCs.
